# Gov.uk Data Cleaning

In [2]:
import numpy as np
import scipy.stats as stats
import seaborn as sns
import matplotlib.pyplot as plt
import pandas as pd
import requests
from bs4 import BeautifulSoup

import pprintpp
pp = pprintpp.PrettyPrinter(indent=4)

plt.style.use('ggplot')

%config InlineBackend.figure_format = 'retina'
%matplotlib inline

# Members Dataframe

In [3]:
members = pd.read_pickle("../Data-Acquisition/members.pkl")
members.head()

id          display_name                            full_title gender  \
0   172       Ms Diane Abbott                Rt Hon Diane Abbott MP      F   
1  3305       Lord Aberconway                   The Lord Aberconway      M   
2  3469  The Duke of Abercorn        His Grace the Duke of Abercorn      M   
3  3468         Lord Aberdare  The Rt Hon. the Lord Aberdare KBE DL      M   
4  3898         Lord Aberdare                     The Lord Aberdare      M   

                         member_from  member_house     membership_start  \
0  Hackney North and Stoke Newington             1  1987-06-11T00:00:00   
1                         Hereditary             2  1953-05-23T00:00:00   
2                         Hereditary             2  1979-06-04T00:00:00   
3                Excepted Hereditary             2  1957-12-18T00:00:00   
4                Excepted Hereditary             2  2009-07-20T00:00:00   

        membership_end membership_end_reason membership_end_reason_notes  \
0                 None                  None                        None   
1  1999-11-11T00:00:00              Excluded                        None   
2  1999-11-11T00:00:00              Excluded                        None   
3  2005-01-23T00:00:00                 Death                        None   
4                 None                  None                        None   

  status_is_active status_description status_notes    status_start_date  \
0             True     Current Member         None  2019-12-12T00:00:00   
1              NaN                NaN          NaN                  NaN   
2              NaN                NaN          NaN                  NaN   
3              NaN                NaN          NaN                  NaN   
4             True     Current Member         None  2009-07-20T00:00:00   

          party party_is_lords_main party_is_lords_spiritual  \
0        Labour                True                    False   
1  Conservative                True                     True   
2  Conservative                True                     True   
3  Conservative                True                     True   
4    Crossbench                True                     True   

  party_is_independent  party_government_type  
0                False                    3.0  
1                False                    0.0  
2                False                    0.0  
3                False                    0.0  
4                False                    NaN

In [4]:
members.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4763 entries, 0 to 4762
Data columns (total 19 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   id                           4763 non-null   int64  
 1   display_name                 4763 non-null   object 
 2   full_title                   4763 non-null   object 
 3   gender                       4763 non-null   object 
 4   member_from                  4760 non-null   object 
 5   member_house                 4763 non-null   int64  
 6   membership_start             4763 non-null   object 
 7   membership_end               3292 non-null   object 
 8   membership_end_reason        2474 non-null   object 
 9   membership_end_reason_notes  313 non-null    object 
 10  status_is_active             1471 non-null   object 
 11  status_description           1471 non-null   object 
 12  status_notes                 6 non-null      object 
 13  status_start_date 

In [5]:
for i in members.columns:
    print(i)

id
display_name
full_title
gender
member_from
member_house
membership_start
membership_end
membership_end_reason
membership_end_reason_notes
status_is_active
status_description
status_notes
status_start_date
party
party_is_lords_main
party_is_lords_spiritual
party_is_independent
party_government_type


### Id

✏️ No duplicates, good. 

In [6]:
members[members['id'].duplicated() == True]

Empty DataFrame
Columns: [id, display_name, full_title, gender, member_from, member_house, membership_start, membership_end, membership_end_reason, membership_end_reason_notes, status_is_active, status_description, status_notes, status_start_date, party, party_is_lords_main, party_is_lords_spiritual, party_is_independent, party_government_type]
Index: []

### Display Name

In [7]:
members.display_name.value_counts()

The Lord Bishop of Gloucester    4
The Lord Bishop of Ely           4
The Lord Bishop of Durham        4
Lord Sinha                       4
The Lord Bishop of Derby         4
                                ..
Lord Goddard of Stockport        1
Dr Norman Godman                 1
Mr Roger Godsiff                 1
Lord Godson                      1
Lord Zuckerman                   1
Name: display_name, Length: 4425, dtype: int64

In [8]:
members[members["display_name"] == "The Lord Bishop of Gloucester"]

id                   display_name  \
1656  2263  The Lord Bishop of Gloucester   
1657  2601  The Lord Bishop of Gloucester   
1658  3906  The Lord Bishop of Gloucester   
1659  4540  The Lord Bishop of Gloucester   

                                      full_title gender member_from  \
1656  The Rt Revd. the Lord Bishop of Gloucester      M     Bishops   
1657   The Rt Rev. the Lord Bishop of Gloucester      M     Bishops   
1658   The Rt Rev. the Lord Bishop of Gloucester      M     Bishops   
1659   The Rt Rev. the Lord Bishop of Gloucester      F     Bishops   

      member_house     membership_start       membership_end  \
1656             2  1981-09-30T00:00:00  1991-10-31T00:00:00   
1657             2  1998-01-31T00:00:00  2003-12-31T00:00:00   
1658             2  2009-10-27T00:00:00  2014-11-21T00:00:00   
1659             2  2015-09-07T00:00:00                 None   

     membership_end_reason membership_end_reason_notes status_is_active  \
1656               Retired                        None              NaN   
1657               Retired                        None              NaN   
1658               Retired                        None              NaN   
1659                  None                        None             True   

     status_description status_notes    status_start_date    party  \
1656                NaN          NaN                  NaN  Bishops   
1657                NaN          NaN                  NaN  Bishops   
1658                NaN          NaN                  NaN  Bishops   
1659     Current Member         None  2015-09-07T00:00:00  Bishops   

     party_is_lords_main party_is_lords_spiritual party_is_independent  \
1656                True                     True                False   
1657                True                     True                False   
1658                True                     True                False   
1659                True                     True                False   

      party_government_type  
1656                    NaN  
1657                    NaN  
1658                    NaN  
1659                    NaN

✏️ These aren't duplicates as they are titles that were passed on from one person to another, so they are almost like job titles rather than unique identifiers. 

### Full Title

In [9]:
members.full_title.value_counts()

The Lord Sinha                          4
His Grace the Duke of Northumberland    3
The Rt Rev. the Lord Bishop of Derby    3
The Lord Wolverton                      3
The Rt Hon. the Earl of Powis           3
                                       ..
Rt Hon John Glen MP                     1
The Rt Hon. the Lord Glenamara CH       1
The Lord Glenarthur DL                  1
The Lord Glenconner                     1
The Lord Zuckerman                      1
Name: full_title, Length: 4523, dtype: int64

In [10]:
members[members["full_title"] == "The Lord Sinha"]

id display_name      full_title gender member_from  member_house  \
3983  2323   Lord Sinha  The Lord Sinha      M  Hereditary             2   
3984  2440   Lord Sinha  The Lord Sinha      M  Hereditary             2   
3985  2947   Lord Sinha  The Lord Sinha      M  Hereditary             2   
3986  3122   Lord Sinha  The Lord Sinha      M  Hereditary             2   

         membership_start       membership_end membership_end_reason  \
3983  1967-05-11T00:00:00  1989-01-06T00:00:00                 Death   
3984  1999-01-18T00:00:00  1999-11-11T00:00:00              Excluded   
3985  1989-01-06T00:00:00  1992-07-25T00:00:00                 Death   
3986  1992-07-25T00:00:00  1999-01-19T00:00:00                 Death   

     membership_end_reason_notes status_is_active status_description  \
3983                        None              NaN                NaN   
3984                        None              NaN                NaN   
3985                        None              NaN                NaN   
3986                        None              NaN                NaN   

     status_notes status_start_date           party party_is_lords_main  \
3983          NaN               NaN  Non-affiliated               False   
3984          NaN               NaN  Non-affiliated               False   
3985          NaN               NaN  Non-affiliated               False   
3986          NaN               NaN  Non-affiliated               False   

     party_is_lords_spiritual party_is_independent  party_government_type  
3983                     True                False                    NaN  
3984                     True                False                    NaN  
3985                     True                False                    NaN  
3986                     True                False                    NaN

✏️ I think the same as above applies here...

### Gender

In [11]:
members.gender.value_counts(dropna=False)

M    3971
F     792
Name: gender, dtype: int64

✏️ No NAs - good, distribution - disappointing. 

### Member from

In [12]:
members.member_from.value_counts()

Life peer                 1378
Hereditary                 911
Excepted Hereditary        150
Bishops                    124
Life Peer (judicial)        50
                          ... 
North East Hampshire         1
Battersea North              1
Orkney and Shetland          1
Plymouth, Devonport          1
Kenilworth and Southam       1
Name: member_from, Length: 931, dtype: int64

In [13]:
members[members['member_from'].isna() == True]

id             display_name                   full_title gender  \
2678  2184  Lord Lucas of Chilworth  The Lord Lucas of Chilworth      M   
3634  3826         The Lord Rennell             The Lord Rennell      M   
4311  3802    Lord Thomson of Fleet        Lord Thomson of Fleet      M   

     member_from  member_house     membership_start       membership_end  \
2678        None             2  2001-11-10T00:00:00  1999-11-11T00:00:00   
3634        None             2  2006-12-09T00:00:00  1999-11-11T00:00:00   
4311        None             2  2006-06-12T00:00:00  1999-11-11T00:00:00   

     membership_end_reason membership_end_reason_notes status_is_active  \
2678              Excluded                        None              NaN   
3634              Excluded                        None              NaN   
4311              Excluded                        None              NaN   

     status_description status_notes status_start_date party  \
2678                NaN          NaN               NaN   NaN   
3634                NaN          NaN               NaN   NaN   
4311                NaN          NaN               NaN   NaN   

     party_is_lords_main party_is_lords_spiritual party_is_independent  \
2678                 NaN                      NaN                  NaN   
3634                 NaN                      NaN                  NaN   
4311                 NaN                      NaN                  NaN   

      party_government_type  
2678                    NaN  
3634                    NaN  
4311                    NaN

✏️ Dropping the NA values... 

In [14]:
members.dropna(subset='member_from', axis=0, inplace=True)
members[members['member_from'].isna() == True]

Empty DataFrame
Columns: [id, display_name, full_title, gender, member_from, member_house, membership_start, membership_end, membership_end_reason, membership_end_reason_notes, status_is_active, status_description, status_notes, status_start_date, party, party_is_lords_main, party_is_lords_spiritual, party_is_independent, party_government_type]
Index: []

### Member house

In [15]:
members.member_house.value_counts(dropna=False)

2    2631
1    2129
Name: member_house, dtype: int64

✏️ No NAs - good.

💡 1: Commons, 2: Lords

### Membership start, end & membership status

In [16]:
print(len(members[members['membership_start'].isna() == True]))
print(len(members[members['membership_end'].isna() == True]))

0
1471


✏️ No NAs for membership start - good. NAs for membership end must mean they are currently active? Checking:

In [17]:
members.status_is_active.value_counts(dropna=False)

NaN      3289
True     1426
False      45
Name: status_is_active, dtype: int64

In [18]:
members.status_description.value_counts(dropna=False)

NaN                 3289
Current Member      1426
Leave of Absence      39
Disqualified           3
Suspended              3
Name: status_description, dtype: int64

In [19]:
members.status_notes.value_counts(dropna=False)

NaN                        3289
None                       1465
Member of the Judiciary       3
Suspended                     3
Name: status_notes, dtype: int64

In [20]:
members.membership_end_reason.value_counts(dropna=False)

None                        2289
Death                       1016
Excluded                     626
Retired                      445
Defeated                     254
Standing Down                 63
General Election              24
Resignation (Northstead)      15
Resignation (Chiltern)        15
Non-attendance                 8
Recall Petition                2
Resigned                       1
Election petition              1
Deselected                     1
Name: membership_end_reason, dtype: int64

In [21]:
members.membership_end_reason_notes.value_counts(dropna=False)

None                                                                                               4447
Dissolution of Parliament                                                                           143
Retired under the House of Lords Reform Act 2014                                                    108
House of Lords Reform Act 2014                                                                        3
Resigned under the Constitutional Reform and Governance Act 2010                                      3
                                                                                                   ... 
Became London Mayor 08/05/16                                                                          1
Retired under the House of Lords Reform Act 2014 [Baroness Lockwood died on 29 April 2019]            1
Retired under the House of Lords Reform Act 2014\n \n[Lord Luke died on 2 October 2015]               1
Mr. Adams refused to accept the office of Crown Steward and Bail

### Adding a Membership Duration Column

In [22]:
members[['membership_start','membership_end']] = members[['membership_start','membership_end']].apply(pd.to_datetime) #if conversion required
members['membership_duration'] = (members['membership_end'] - members['membership_start']).dt.days

In [23]:
members['membership_duration'].isna().value_counts()

False    3289
True     1471
Name: membership_duration, dtype: int64

In [24]:
members.status_is_active.value_counts(dropna=False)

NaN      3289
True     1426
False      45
Name: status_is_active, dtype: int64

💡 In status is active NaN means the membership has ended, True means the membership is active, and False means the member is either on leave of Absence, Disqualified or Suspended... adjusting the labels accordingly & computing the active members' current membership durations:

In [25]:
members = members.rename(columns={"status_is_active": "membership_status"})

members.loc[members['membership_status'] == True, 'membership_status'] = "Active"
members.loc[members['membership_status'] == False, 'membership_status'] = "TemporarilyInactive"
members.loc[members['membership_status'].isna(), 'membership_status'] = "PermanentlyInactive"

members = members.drop("status_description", axis=1)

members['membership_status'].value_counts()

PermanentlyInactive    3289
Active                 1426
TemporarilyInactive      45
Name: membership_status, dtype: int64

In [26]:
members['membership_end'].fillna(pd.to_datetime('today').normalize(), inplace=True)
members['membership_duration'] = (members['membership_end'] - members['membership_start']).dt.days
members = members.rename(columns={"membership_duration": "membership_duration_in_days"})

members.membership_duration_in_days.isna().value_counts()

False    4760
Name: membership_duration_in_days, dtype: int64

In [27]:
members['member_since'] = members['membership_start'].dt.year
members['member_till'] = members['membership_end'].dt.year

members = members.drop(["membership_start", "membership_end"], axis=1)
members.head()

id          display_name                            full_title gender  \
0   172       Ms Diane Abbott                Rt Hon Diane Abbott MP      F   
1  3305       Lord Aberconway                   The Lord Aberconway      M   
2  3469  The Duke of Abercorn        His Grace the Duke of Abercorn      M   
3  3468         Lord Aberdare  The Rt Hon. the Lord Aberdare KBE DL      M   
4  3898         Lord Aberdare                     The Lord Aberdare      M   

                         member_from  member_house membership_end_reason  \
0  Hackney North and Stoke Newington             1                  None   
1                         Hereditary             2              Excluded   
2                         Hereditary             2              Excluded   
3                Excepted Hereditary             2                 Death   
4                Excepted Hereditary             2                  None   

  membership_end_reason_notes    membership_status status_notes  \
0                        None               Active         None   
1                        None  PermanentlyInactive          NaN   
2                        None  PermanentlyInactive          NaN   
3                        None  PermanentlyInactive          NaN   
4                        None               Active         None   

     status_start_date         party party_is_lords_main  \
0  2019-12-12T00:00:00        Labour                True   
1                  NaN  Conservative                True   
2                  NaN  Conservative                True   
3                  NaN  Conservative                True   
4  2009-07-20T00:00:00    Crossbench                True   

  party_is_lords_spiritual party_is_independent  party_government_type  \
0                    False                False                    3.0   
1                     True                False                    0.0   
2                     True                False                    0.0   
3                     True                False                    0.0   
4                     True                False                    NaN   

   membership_duration_in_days  member_since  member_till  
0                        13095          1987         2023  
1                        16973          1953         1999  
2                         7465          1979         1999  
3                        17203          1957         2005  
4                         5020          2009         2023

In [28]:
members.membership_end_reason.fillna("None", inplace=True)

In [29]:
members = members.drop(["membership_end_reason_notes", "status_notes", "status_start_date"], axis=1)

In [30]:
members.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4760 entries, 0 to 4762
Data columns (total 16 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   id                           4760 non-null   int64  
 1   display_name                 4760 non-null   object 
 2   full_title                   4760 non-null   object 
 3   gender                       4760 non-null   object 
 4   member_from                  4760 non-null   object 
 5   member_house                 4760 non-null   int64  
 6   membership_end_reason        4760 non-null   object 
 7   membership_status            4760 non-null   object 
 8   party                        4760 non-null   object 
 9   party_is_lords_main          4760 non-null   object 
 10  party_is_lords_spiritual     4760 non-null   object 
 11  party_is_independent         4760 non-null   object 
 12  party_government_type        3031 non-null   float64
 13  membership_duratio

### Party

In [31]:
members.party.value_counts(dropna=False)

Conservative                        1779
Labour                              1194
Crossbench                           652
Liberal Democrat                     273
Other                                233
Non-affiliated                       136
Bishops                              124
Scottish National Party               79
Independent                           66
Labour (Co-op)                        58
Democratic Unionist Party             22
Social Democratic Party               22
Ulster Unionist Party                 21
Sinn Féin                             14
Liberal                               12
Plaid Cymru                           11
Independent Labour                     9
Social Democratic & Labour Party       8
Independent Conservative               7
Not known                              5
The Independent Group for Change       5
Green Party                            4
Alliance                               3
Alba Party                             2
Independent Soci

✏️ Dropping NAs:

In [32]:
members = members[members.party.isna() == False]

### Party_is_lords_main / party_is_lords_spiritual / party_is_independent / party_government_type

In [33]:
members.party_is_lords_main.value_counts(dropna=False)

members.loc[members['party_is_lords_main'] == False, 'party_is_lords_main'] = 0
members.loc[members['party_is_lords_main'] == True, 'party_is_lords_main'] = 1

members.party_is_lords_main.value_counts()

1    4086
0     674
Name: party_is_lords_main, dtype: int64

In [34]:
members.loc[members['party_is_lords_spiritual'] == False, 'party_is_lords_spiritual'] = 0
members.loc[members['party_is_lords_spiritual'] == True, 'party_is_lords_spiritual'] = 1

members.party_is_lords_spiritual.value_counts()

1    2736
0    2024
Name: party_is_lords_spiritual, dtype: int64

In [35]:
members.party_is_independent.value_counts(dropna=False)

members.loc[members['party_is_independent'] == False, 'party_is_independent'] = 0
members.loc[members['party_is_independent'] == True, 'party_is_independent'] = 1

members.party_is_independent.value_counts()

0    4694
1      66
Name: party_is_independent, dtype: int64

In [36]:
members.party_government_type.value_counts(dropna=False)

0.0    1779
NaN    1729
3.0    1252
Name: party_government_type, dtype: int64

💡 No idea what this means - dropping.

In [37]:
del members["party_government_type"]

members.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4760 entries, 0 to 4762
Data columns (total 15 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   id                           4760 non-null   int64 
 1   display_name                 4760 non-null   object
 2   full_title                   4760 non-null   object
 3   gender                       4760 non-null   object
 4   member_from                  4760 non-null   object
 5   member_house                 4760 non-null   int64 
 6   membership_end_reason        4760 non-null   object
 7   membership_status            4760 non-null   object
 8   party                        4760 non-null   object
 9   party_is_lords_main          4760 non-null   object
 10  party_is_lords_spiritual     4760 non-null   object
 11  party_is_independent         4760 non-null   object
 12  membership_duration_in_days  4760 non-null   int64 
 13  member_since                 4760

In [38]:
members.to_pickle("cleaned_members_gov_uk.pkl")

# Votes Dataframe

In [39]:
votes = pd.read_pickle("../Data-Acquisition/votes.pkl")
votes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 731949 entries, 0 to 731948
Data columns (total 7 columns):
 #   Column             Non-Null Count   Dtype 
---  ------             --------------   ----- 
 0   member_id          731949 non-null  object
 1   vote_title         731949 non-null  object
 2   vote_date          731949 non-null  object
 3   aye_count          731949 non-null  object
 4   no_count           731949 non-null  object
 5   member_voted_aye   731949 non-null  object
 6   member_was_teller  731949 non-null  object
dtypes: object(7)
memory usage: 39.1+ MB


In [40]:
votes.head()

member_id                                         vote_title  \
0       172  Finance (No. 2) Bill: Reasoned Amendment to Se...   
1       172  Illegal Migration Bill: Committee of the whole...   
2       172  Illegal Migration Bill: Committee of the whole...   
3       172  Illegal Migration Bill: Committee of the whole...   
4       172  Illegal Migration Bill: Committee of the whole...   

             vote_date aye_count no_count member_voted_aye member_was_teller  
0  2023-03-29T18:03:00       211      289             True             False  
1  2023-03-28T20:37:00       248      301             True             False  
2  2023-03-28T20:25:00       249      301             True             False  
3  2023-03-28T20:13:00       248      299             True             False  
4  2023-03-28T20:01:00       302      242            False             False

### Vote Years

In [41]:
votes['vote_date'] = votes['vote_date'].apply(pd.to_datetime) 
votes['vote_year'] = votes['vote_date'].dt.year

votes['vote_year'].unique()

array([2023, 2022, 2021, 2020, 2019, 2018, 2017, 2016])

✏️ We have voting data spanning 8 years from 2016-2023.

### Vote Titles

In [42]:
len(votes['vote_title'].unique())

1450

✏️ There are a total of 1450 votes within this dataset.

✏️ Assigning unique id's:

In [43]:
votes['vote_id'] = votes.groupby(['vote_title']).ngroup()

print(len(votes['vote_title'].unique()))
print(len(votes['vote_id'].unique()))

votes[["vote_title", "vote_id"]].sort_values(by="vote_id")

1450
1450


vote_title  vote_id
120902  \tElections Bill: Government Motion to disagre...        0
236829  \tElections Bill: Government Motion to disagre...        0
641185  \tElections Bill: Government Motion to disagre...        0
501855  \tElections Bill: Government Motion to disagre...        0
10274   \tElections Bill: Government Motion to disagre...        0
...                                                   ...      ...
621768       s5 European Communities (Amendment) Act 1993     1449
110277       s5 European Communities (Amendment) Act 1993     1449
365890       s5 European Communities (Amendment) Act 1993     1449
497072       s5 European Communities (Amendment) Act 1993     1449
342855       s5 European Communities (Amendment) Act 1993     1449

[731949 rows x 2 columns]

I'll NLP the vote_titles closer to the modelling stage to save memory...

### Member was teller

Dropping this as it will be too strong a predictor ...

In [44]:
del votes["member_was_teller"]

### Member voted aye

In [45]:
votes.loc[votes['member_voted_aye'] == False, 'member_voted_aye'] = 0
votes.loc[votes['member_voted_aye'] == True, 'member_voted_aye'] = 1

votes.member_voted_aye.value_counts()

1    377589
0    354360
Name: member_voted_aye, dtype: int64

✏️ Pretty even classes, great :)

### Adding a vote outcome column

In [46]:
comparison_column = np.where(votes["aye_count"] > votes["no_count"], 1, 0)
votes["vote_outcome"] = comparison_column

In [47]:
votes.head()

member_id                                         vote_title  \
0       172  Finance (No. 2) Bill: Reasoned Amendment to Se...   
1       172  Illegal Migration Bill: Committee of the whole...   
2       172  Illegal Migration Bill: Committee of the whole...   
3       172  Illegal Migration Bill: Committee of the whole...   
4       172  Illegal Migration Bill: Committee of the whole...   

            vote_date aye_count no_count member_voted_aye  vote_year  vote_id  \
0 2023-03-29 18:03:00       211      289                1       2023      495   
1 2023-03-28 20:37:00       248      301                1       2023      683   
2 2023-03-28 20:25:00       249      301                1       2023      680   
3 2023-03-28 20:13:00       248      299                1       2023      677   
4 2023-03-28 20:01:00       302      242                0       2023      679   

   vote_outcome  
0             0  
1             0  
2             0  
3             0  
4             1

✏️ Vote outcome = 1 if more ayes than nos, else 0.

## member_id

In [48]:
len(votes.member_id.unique())

901

✏️ 901 unique members in this dataset. Good, saving it to a cleaned file.

In [49]:
votes.to_pickle("cleaned_votes.pkl")

# Written Statements

In [50]:
statements = pd.read_pickle("../Data-Acquisition/written_statements.pkl")
statements.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6255 entries, 0 to 6254
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   statement_id         6255 non-null   int64 
 1   statement_title      6255 non-null   object
 2   statement_text       6255 non-null   object
 3   member_id            6255 non-null   int64 
 4   member_name          0 non-null      object
 5   member_role          6255 non-null   object
 6   statement_date       6255 non-null   object
 7   answering_body_id    6255 non-null   int64 
 8   answering_body_name  6255 non-null   int64 
 9   house                6255 non-null   object
dtypes: int64(4), object(6)
memory usage: 488.8+ KB


✏️ Dropping empty column member_name: 

In [51]:
statements = statements.drop(columns="member_name")

💡 Parking this for the MVP - I think it's too much NLP work for its potential reward.